blob-clients-common/blob-jersey2-client-common/src/main/java/com/bazaarvoice/emodb/blob/jersey2/client/BlobStoreJersey2Client.java

```diff
@@ -109,24 +118,43 @@ public BlobStoreJersey2Client(URI endPoint, EmoClient client,
             _connectionManagementService = Executors.newSingleThreadScheduledExecutor(
                     new ThreadFactoryBuilder().setNameFormat("blob-store-client-connection-reaper-%d").build());
         }
+
+        _log.info("creating retry policy with {},{},{}", retryPolicy.getConfig().getMaxRetries(),
```

Comment:
https://github.com/bazaarvoice/emodb/pull/437/files#diff-ff4b5d285b67f3d64476b4bc3e518fe60aa41587706968ae0e2ad47cf69f6e26R122-R139 should be extracted.

policy that is provided as method parameter should be used without modifications.

Reply:
Two things wanted to add here. 

1. Why we're creating RetryPolicy object instead of using one that is sent as parameter.

    The usage of .onFailure(), .onRetry() can either be while building the retry policy (line 126 to 139) OR while 
    sending client request, like this
    `Failsafe.with(_retryPolicy)
                .onFailure(e -> {// add here})
                .onRetry(e -> {//add here})
                .run(() -> _client.resource(uri)
                .type(MediaType.APPLICATION_JSON_TYPE)
                .header(ApiKeyRequest.AUTHENTICATION_HEADER, apiKey)
                .put(attributes));`
       
     Can not set these on an existing retry policy directly. 
       Hence instead of adding .onFailure() & .onRetry() calls in each http request and to minimize repetition, I've 
       created RetryPolicy object by borrowing configs sent in the parameter. 

2. Why can't we set these params [.onRetry(), onFailure()] in clientFactory where we're building our retryPolicy first time. 

    In Client class, we have` convertException()`method to covert the EmoResponse to relevant exception. we're using this with onFailure() call (line 137). We can't utilize this method if onFailure() is set in ClientFactory.

Change option: 

Option 1: Reuse the RetryPolicy object sent as parameter and set .onRetry(), .onFailure() in each http request with FailSafe.with() call. 
(RetryPolicy object created 1 time, but both the above methods will be repeated number of times = number of httpRequests for each API client)

Option 2: Move createRetryPolicy() to Client class instead of ClientFactory. 
(RetryPolicy object created 1 time, but maxRetryAttempt, minDelay, maxDelay params needs to be sent to Client class)
Option 3:  keep the existing change, but extract creating _only_ defaultRetryPolicy to Client instead of ClientFactory

Open to other suggestions

Reply:
created draft commit - https://github.com/bazaarvoice/emodb/pull/437/commits/cca208f28bf645d22ae63b9fb08d09c82ac6ed68 to show the approach:
- `BlobStoreJersey2Client` is the lowest level, it shouldn't create a retry policy and just use it (passed via constructor)
- in `AbstractBlobStoreJersey2ClientFactory`  public methods - `createRetryPolicy` and `isRetriableException` are introduced so end-users are able to override default behaviour

hope, it makes sense
consider introducing unit tests like https://github.com/failsafe-lib/failsafe/issues/79